lib/internal/streams/end-of-stream.js

```diff
@@ -13,6 +13,15 @@ function isRequest(stream) {
   return stream.setHeader && typeof stream.abort === 'function';
 }
 
+function isOutgoingMessage(stream) {
+  return (
+    typeof stream._removedConnection === 'boolean' &&
+    typeof stream._removedContLen === 'boolean' &&
+    typeof stream._removedTE === 'boolean' &&
+    typeof stream._closed === 'boolean'
+  );
+}
```

Comment:
I'm not really fond of this approach, it relies too much on internal variables, would this work across node versions?

Reply:
Any better alternative? I'm not sure when `_closed` was introduced but the rest should be ok across node version.

Reply:
Great!